tests/Doctrine/MongoDB/Tests/CollectionEventsChangingContextTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Doctrine\MongoDB\Tests;
+
+use Doctrine\Common\EventManager;
+use Doctrine\MongoDB\Events;
+use Doctrine\MongoDB\Event\AggregateEventArgs;
+use Doctrine\MongoDB\Event\DistinctEventArgs;
+use Doctrine\MongoDB\Event\EventArgs;
+use Doctrine\MongoDB\Event\FindEventArgs;
+use Doctrine\MongoDB\Event\GroupEventArgs;
+use Doctrine\MongoDB\Event\MapReduceEventArgs;
+use Doctrine\MongoDB\Event\MutableEventArgs;
+use Doctrine\MongoDB\Event\NearEventArgs;
+use Doctrine\MongoDB\Event\UpdateEventArgs;
+
+class CollectionEventsTest extends \PHPUnit_Framework_TestCase
+{
+    private $database;
+    private $mongoCollection;
+
+    public function setUp()
+    {
+        $this->database = $this->getMockDatabase();
+        $this->mongoCollection = $this->getMockMongoCollection();
+    }
+
+    public function testAggregate()
```

Comment:
Sorry I didn't get around to commenting. Some people like to use mocks, some don't - I belong in the latter camp. That being said, the tests work fine this way and do their job. :+1: